hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/impl/ERCTransferPrecompile.java

```diff
@@ -154,9 +197,15 @@ public void run(final MessageFrame frame) {
         }
 
         if (isFungible) {
-            frame.addLog(getLogForFungibleTransfer(asTypedEvmAddress(tokenID)));
+            frame.addLog(
+                    getLogForFungibleTransfer(
+                            asTypedEvmAddress(
+                                    tokenID != null ? tokenID : extractTokenIdFromSyntheticOp())));
```

Comment:
Do this `asTypedEvmAddress` calculation and the one at line 207-298 once at line 198.

Reply:
Fixed